src/main/java/org/itsallcode/holidays/calculator/logic/conditions/Condition.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * holiday-calculator
+ * Copyright (C) 2021 itsallcode <github@kuhnke.net>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.itsallcode.holidays.calculator.logic.conditions;
+
+import java.time.MonthDay;
+import java.time.Year;
+
+public abstract class Condition {
+
+	public static final Condition APPLIES_ALWAYS = new ConstantCondition(true);
+
+	public static final Condition not(Condition other) {
+		return other.copy().negate();
+	}
+
+	public abstract boolean applies(Year year);
+
+	public abstract String toString(String prefix);
+
+	protected abstract Condition copy();
+
+	protected boolean negate;
```

Comment:
Would it be possible to make this immutable? Then we don't need the copy method, and `negate()` would return a new instance.
```suggestion
	protected final boolean negate;
```

In general I would prefer composition to inheritance. You could implement this by having a `NegatedCondition` that always returns the negation of a delegate `Condition` in it's `applies()` method.

Reply:
Understood. Let my try.